src/software/jetson_nano/display/robot_ui.py

```diff
@@ -0,0 +1,154 @@
+    def __init__(self):
+
+        # Initialize redis server and our redis dictionary
+        self.r = redis.Redis(host="localhost", port=6379, db=0)
+        self.redis_dict = {}
+        for key in redis_keys:
+            self.redis_dict[key] = float(self.r.get(key).decode("UTF-8"))
+        self.shutdown = False  # This flag will be used to stop polling redis
+
+        # Draw Tbots logo on first boot
+        self.lcd_display = LcdDisplay()
+        self.lcd_display.draw_image("./lcd_user_interface/imgs/tbots.jpg")
+        self.curr_screen = "Home"
+
+        # All of our screens
+        self.screens = {
+            "Home": HomeScreen(self.lcd_display, self.redis_dict, status_codes),
+            "Menu": MenuScreen(self.lcd_display, status_codes),
+            "Wheels": WheelsScreen(self.lcd_display, self.redis_dict, status_codes),
+            "Chip and Kick": ChipAndKickScreen(
+                self.lcd_display, self.redis_dict, status_codes
+            ),
+        }
+
+        def on_click():
+            """ Execute on click callback of curr screen """
+            action = self.screens[self.curr_screen].on_click()
+
+            if status_codes["change screen"] in action:
+                self.curr_screen = action[status_codes["change screen"]]
+                self.screens[self.curr_screen].draw_screen()
+                self.lcd_display.show()
+            elif status_codes["update redis"] in action:
+                payload = action[status_codes["update redis"]]
+                self.r.set(payload["redis key"], payload["value"])
+                self.redis_dict[payload["redis key"]] = payload["value"]
+                # print("Key: {}, Value: {}".format(payload["redis key"], self.r.get(payload["redis key"]).decode("UTF-8")))
+
+        def on_clockwise_rotate():
+            """ Execute the clockwise rotate callback of curr screen """
+            self.screens[self.curr_screen].on_clockwise_rotate()
+
+        def on_counterclockwise_rotate():
+            """ Execute the counterclockwise rotate callback of curr screen """
+            self.screens[self.curr_screen].on_counterclockwise_rotate()
+
+        self.rotary_encoder = RotaryEncoder(
+            PIN_1,
+            PIN_2,
+            BUTTON_PIN,
+            on_clockwise_rotate,
+            on_counterclockwise_rotate,
+            on_click,
+        )
+
+        self.rotary_encoder.start()
+
+    def poll_redis(self):
+        """ Update redis dict every 10 seconds """
+        while not self.shutdown:
+            for key in redis_keys:
+                self.redis_dict[key] = float(self.r.get(key).decode("UTF-8"))
+
+            for _, screen in self.screens.items():
+                if _ != "Menu":
+                    screen.update_values(self.redis_dict)
+            time.sleep(10)
+
+    def stop(self):
+        """ Cleanup the GPIO pins """
+        self.shutdown = True
+        self.rotary_encoder.stop()
+
+
+# For testing
+if __name__ == "__main__":
```

Comment:
should this be pulled into a separate test file?

Reply:
does `py_library` let you run this? 

i.e can you do `bazel  run //software/jetson_nano/display:robot_ui`